examples/auto_instrumentation/README.md

```diff
@@ -0,0 +1,119 @@
+# Overview
+
+This example shows how to use auto-instrumentation in OpenTelemetry. This example is also based on a previous example
+for OpenTracing that can be found [here](https://github.com/yurishkuro/opentracing-tutorial/tree/master/python).
+
+This example uses 2 scripts whose main difference is they being instrumented manually or not:
+
+1. `publisher_instrumented.py` which has been instrumented manually
+2. `publisher_uninstrumented.py` which has not been instrumented manually
+
+The former will be run without the automatic instrumentation agent and the latter with the automatic instrumentation
+agent. They should produce an equal result, showing that the automatic instrumentation agent does the equivalent
+of what manual instrumentation does.
+
+In order to understand this better, here is the relevant part of both scripts:
+
+## Publisher instrumented manually
+
+`publisher_instrumented.py`
+
+```python
+@app.route("/publish_request")
+def publish_request():
+
+    with tracer.start_as_current_span(
+        "publish_request", propagators.extract(get_as_list, request.headers)
+    ):
+        hello_str = request.args.get("helloStr")
+        print(hello_str)
+        return "published"
+```
+
+## Publisher not instrumented manually
+
+`publisher_uninstrumented.py`
+
+```python
+@app.route("/publish_request")
+def publish_request():
+    hello_str = request.args.get("helloStr")
+    print(hello_str)
+    return "published"
+```
+
+# Preparation
+
+This example will be executed in a separate virtual environment:
+
+```sh
+$ mkdir auto_instrumentation
+$ virtualenv auto_instrumentation
+$ source auto_instrumentation/bin/activate
+```
+
+# Installation
+
+```sh
+$ git clone git@github.com:open-telemetry/opentelemetry-python.git
+$ cd opentelemetry-python
+$ pip3 install -e opentelemetry-api
+$ pip3 install -e opentelemetry-sdk
+$ pip3 install -e opentelemetry-auto-instrumentation
+$ pip3 install -e ext/opentelemetry-ext-flask
+$ pip3 install flask
+$ pip3 install requests
+```
+
+# Execution
+
+## Execution of the manually instrumented publisher
+
+This is done in 3 separate consoles, one to run each of the scripts that make up this example:
+
+```sh
+$ source auto_instrumentation/bin/activate
```

Comment:
Sorry if I missed something. What is the purpose of `formatter.py`? Aren't the other files sufficient enough for an example?